### DISCUSSION TITLE
3-pin SK6812 / Level Shifter Required?

### DISCUSSION THREAD

**Discussion by jpleasants** [awawa-dev/HyperSPI#31] [Q&A] [answered]
I intend to go with a single-segment setup with Pi4 and ESP32 using HyperSPI. All components are less than 8cm away from each other. Pi4 is powered separately from LEDs. Plan on connecting Pi4 GND to ESP32 GND and then ESP32 GND to GND of LED PSU.

Since I have a 3-pin SK6812NW (229 LEDs), are all connections from Pi4 to ESP32 still valid?

![image](https://github.com/awawa-dev/HyperSPI/assets/18755133/b53e6b6f-4a50-4f2d-ab26-3fc1b712e3e1)

Also, will a level shifter be required or can I just connect data line of ESP32 directly to DIN of SK6812?

**Comment by awawa-dev** [comment, marked as answer]
Hi
The connection from RPi to ESP in HyperSPI does not depend on the type of LEDs used, only on the specific ESP model. If you took the diagram from the project's home page or from the last blog post, it should be OK. If you use a different ESP model than the one found there, you must make sure of the correct pinout yourself.
Since ESP and RPi have a pinout operating in the same 3.3V range, you don't need a level shifter between them. But since LEDs usually operate at a higher voltage for communication, a level shifter must be used because even taking into account the tolerance level, for 5V LEDs the RPi/ESP voltage of 3.3V is slightly below this limit (though it really depends on the supplied VCC, which may vary in in practice from the ideal 5V but we should not rely on this).

**Reply by jpleasants** [reply]
Thank you, that answers my questions!
